### ISSUE TITLE
mT5 fine-tune for en-my got "NaN" in training loss and validation loss

### ISSUE THREAD

**Issue by learnercat** [huggingface/notebooks#31]
I tried **to fine-tune mT5 for English->Myanmar translation** from [Tatoeba-Challenge](https://github.com/Helsinki-NLP/Tatoeba-Challenge) Dataset. I followed to train [this notebook example](https://github.com/huggingface/notebooks/blob/master/examples/translation.ipynb) of en-ro translation.  And I used model_checkpoint as "google/mt5-small". I tested 1~4 epoch training. 
The following is the training parameters, I reduced the batch_size as 4.

`batch_size=4`
`args = Seq2SeqTrainingArguments(`
`  "mt5-translate-en-my",`
`  evaluation_strategy = "epoch",`
`  learning_rate=2e-5,`
`   per_device_train_batch_size=batch_size,`
`   per_device_eval_batch_size=batch_size,`
`weight_decay=0.01,`
`    save_total_limit=3,`
`    num_train_epochs=1,`
`predict_with_generate=True,`
`   fp16=True,
    )`

I got "NaN" in training loss and validation loss as below:

<img width="989" alt="mt5_error" src="https://user-images.githubusercontent.com/25918640/115986186-9c186080-a5e1-11eb-9a74-656bb5185443.png">

Can you please direct me how do I solve it? Thanks in advance.

**Comment by msaroufim** [comment]
What kind of hardware are you using? Do you get the same issue if you set `fp16=False`?

**Comment by Majdoddin** [comment]
@msaroufim Thank you very much, it worked for me on colab. Also the warning `lr_schedual.ster()` before `optimizer.step()` disappeared.
But why should set `fp16=False` even when I have A100 GPU on colab?
